metadata-editor/app/controllers/EditsController.scala

```diff
@@ -113,7 +109,9 @@ object EditsController extends Controller with ArgoHelpers {
           .setAdd(id, "labels", labels)
           .map(publish(id))
           .map(edits => labelsCollection(id, edits.labels.toSet))
-          .map {case (uri, labels) => respondCollection(labels)}
+          .map {case (uri, labels) => respondCollection(labels)} recover {
+          case _ => BadRequest
```

Comment:
Can we be more specific on what we want to recover from? Is there an Amazon exception class we can catch?

Being too broad means we may mistakenly catch unrelated errors and mask them from our logs and attention (e.g. network down, DNS gone, out of memory, disk full, etc).